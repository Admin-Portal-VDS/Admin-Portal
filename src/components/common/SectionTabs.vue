<template>
  <div>
    <div class="navbar">
      <div
        :class="['navbar-item', { active: selected === 'user' }]"
        @click="selectTab('user')"
      >
        Users
      </div>
      <div
        :class="['navbar-item', { active: selected === 'auditLog' }]"
        @click="selectTab('auditLog')"
      >
        Audit Log
      </div>
    </div>
    <hr class="underline" />
    <SearchFilterBar />
  </div>
</template>

<script>
import SearchFilterBar from './SearchFilterBar.vue';

export default {
  name: 'SectionTabs',
  components: { SearchFilterBar },
  data() {
    return {
      selected: 'user',
    };
  },
  methods: {
    selectTab(tab) {
      this.selected = tab;
    },
  },
};
</script>

<style scoped>
.navbar {
  display: flex;
  justify-content: flex-start;
  gap: 20px;
}
.navbar-item {
  cursor: pointer;
  padding: 5px 2px;
  font-size: 1.1rem;
  text-align: center;
  position: relative;
  display: inline-block;
}

.navbar-item.active {
  font-weight: 600;
}

.navbar-item::after {
  content: '';
  position: absolute;
  bottom: -4px;
  left: 0;
  right: 0;
  height: 3px;
  background-color: transparent;
  transition: background-color 0.3s ease;
}

.navbar-item.active::after {
  background-color: black; /* Underline color when active */
}
.underline {
  margin-top: 4px;
}
</style>
