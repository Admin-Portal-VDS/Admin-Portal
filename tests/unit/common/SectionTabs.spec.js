import { mount } from '@vue/test-utils';
import SectionTabs from '@/components/common/SectionTabs.vue';

describe('SectionTabs.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(SectionTabs, {
      propsData: {
        sectionTabs: [
          { name: 'tab1', label: 'Tab 1' },
          { name: 'tab2', label: 'Tab 2' },
        ],
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('matches snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
