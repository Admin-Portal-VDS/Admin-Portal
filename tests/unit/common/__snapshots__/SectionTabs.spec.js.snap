// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionTabs.vue matches snapshot 1`] = `
<div>
  <div class="navbar">
    <div class="navbar-item active"> Users </div>
    <div class="navbar-item"> Audit Log </div>
  </div>
  <hr class="underline">
  <div class="search-bar">
    <div class="top-row">
      <div class="left-section"><button class="search-button"><img src="" alt="Search" class="search-icon"><span>Search</span><input class="search-input" placeholder="" style="display: none;"></button><button class="advanced-filter-button"><img src="" alt="Filter" class="filter-icon"> Advanced Filters </button></div>
      <div class="right-section"><button class="refresh-button"><img src="" alt="Refresh" class="refresh-icon"></button><button class="print-button"><img src="" alt="Print" class="print-icon"></button>
        <div class="export-excel"><button class="simple-button">Export to Excel</button></div>
        <div class="dropdown"><button data-id="dropdown-button" class="dropdown-button">Bulk <span class="arrow"></span></button>
          <!--v-if-->
        </div>
        <div class="checkbox-dropdown"><button data-id="dropdown-button" class="dropdown-button">Show/Hide Columns <span class="arrow"></span></button>
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class="bottom-row">
      <div class="change-user-button"><button class="simple-button">Change Super User</button></div>
      <div class="add-new-button"><button class="simple-button">Add New</button></div>
    </div>
  </div>
  <div class="table-wrapper">
    <table class="table-container">
      <thead>
        <tr>
          <th data-sort="" data-id="name">Name</th>
          <th data-sort="" data-id="age">Age</th>
        </tr>
      </thead>
      <tbody>
        <tr class="table-data">
          <td data-id="0-name">Alice</td>
          <td data-id="0-age">30</td>
        </tr>
        <tr class="table-data">
          <td data-id="1-name">Bob</td>
          <td data-id="1-age">25</td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
